src/dcommands/kick.js

```diff
@@ -12,56 +12,72 @@ class KickCommand extends BaseCommand {
                 break;
-        }
+                case 3: //User doesnt have perms
+                    response = `You don't have permission to kick users!`;
+                    break;
+                case 4: //User cannot kick target
+                    response = `You don't have permission to kick ${target.username}!`;
+                    break;
+                default: //Error occurred
+                    response = `Failed to kick the user! Please check your permission settings and command and retry. \nIf you still can't get it to work, please report it to me by doing \`b!report <your issue>\` with the following:\`\`\`\n${state.message}\n${state.response}\`\`\``;
+                    break;
+            }
 
-        bu.send(msg, response);
+            bu.send(msg, response);*/
+            
+            bu.send(await this.kick(msg, user, reason));
+            
+        } else bu.send(msg, `You didn't tell me who to kick!`);
     }
 
-    async kick(msg, target, reason, tag = false, noPerms = false) {
-        if (!msg.channel.guild.members.get(bot.user.id).permissions.json.kickMembers)
-            return 1;
+    async kick(msg, user, reason, tag = false, noPerms = false) {
+        if (!msg.channel.guild.members.get(bot.user.id).permissions.json.kickMembers) {
+            return [`I don't have permission to kick users!`, '`Bot has no permissions`'];
+        }
         let kickPerms = await bu.guildSettings.get(msg.guild.id, 'kickoverride') || 0;
-        if (!noPerms && !bu.comparePerms(msg.member, kickPerms) && !msg.member.permissions.json.kickMembers)
-            return 3;
-
-        var botPos = bu.getPosition(msg.channel.guild.members.get(bot.user.id));
-        var userPos = bu.getPosition(msg.member);
-        var targetPos = bu.getPosition(msg.channel.guild.members.get(target.id));
-        if (targetPos >= botPos)
-            return 2;
-        if (!noPerms && targetPos >= userPos && msg.author.id != msg.guild.ownerID)
-            return 4;
+        if (!noPerms && !bu.comparePerms(msg.member, kickPerms) && !msg.member.permissions.json.kickMembers) {
+            return [`You don't have permission to ban users!`, '`User has no permissions`'];
+        }
+        
+        let member = msg.guild.members.get(user.id);
+        
+        if(member) {
+            let botPos = bu.getPosition(msg.channel.guild.members.get(bot.user.id));
+            let userPos = bu.getPosition(msg.member);
+            let targetPos = bu.getPosition(msg.channel.guild.members.get(user.id));
+            if (targetPos >= botPos) {
+                return [`I don't have permission to kick ${user.username}!`, '`Bot has no permissions`'];
+            }
+            if (!noPerms && targetPos >= userPos && msg.author.id != msg.guild.ownerID) {
+                return [`You don't have permission to kick ${user.username}!`, '`User has no permissions`'];
+            }
+        }

```

Comment:
I would change this. You can ban someone who isnt on the guild, but not kick them. This block in the ban command is to check if you are able to ban them if they are on the guild. I think just erroring if they arent is a better idea for kick rather than getting an error thrown below.
```suggestion
        let member = msg.guild.members.get(user.id);
        
        if(!member) {
            return [`${user.username} isnt on this server!`, '`User not in guild`'];
        }
        let botPos = bu.getPosition(msg.channel.guild.members.get(bot.user.id));
        let userPos = bu.getPosition(msg.member);
        let targetPos = bu.getPosition(msg.channel.guild.members.get(user.id));
        if (targetPos >= botPos) {
            return [`I don't have permission to kick ${user.username}!`, '`Bot has no permissions`'];
        }
        if (!noPerms && targetPos >= userPos && msg.author.id != msg.guild.ownerID) {
            return [`You don't have permission to kick ${user.username}!`, '`User has no permissions`'];
        }

```